applications/DEMApplication/custom_strategies/strategies/explicit_solver_continuum.cpp

```diff
@@ -904,21 +904,10 @@ namespace Kratos {
 
         ConditionsArrayType& pConditions = GetFemModelPart().GetCommunicator().LocalMesh().Conditions();
         const ProcessInfo& r_process_info = GetFemModelPart().GetProcessInfo();
-        Vector rhs_cond;
-        std::vector<unsigned int> condition_partition;
-        OpenMPUtils::CreatePartition(mNumberOfThreads, pConditions.size(), condition_partition);
-
-        #pragma omp parallel for private (rhs_cond)
-        for (int k = 0; k < mNumberOfThreads; k++) {
-            ConditionsArrayType::iterator it_begin = pConditions.ptr_begin() + condition_partition[k];
-            ConditionsArrayType::iterator it_end = pConditions.ptr_begin() + condition_partition[k + 1];
-
-            for (ConditionsArrayType::iterator it = it_begin; it != it_end; ++it) {
-                it->FinalizeSolutionStep(r_process_info);
-            }
-        }
 
+        block_for_each(pConditions, [&](ModelPart::ConditionType& rCondition){
+            rCondition.FinalizeSolutionStep(r_process_info);
```

Comment:
btw this already exists here: https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/utilities/entities_utilities.h

Reply:
Is it required to pass r_process_info in [&] or is it a recommendation?

Reply:
`[&]` captures everything (by reference), `[&r_current_process_info]` captures on the processinfo (by reference)
In general I would recommend to only capture what you need, but this is minor IMO

If you want to know more I recommend to read up on c++ lambda captures